

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Data Inputs &mdash; TumorDecon 0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=2709fde1"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Signature Matrix Creation" href="sigmatrix.html" />
    <link rel="prev" title="Quickstart" href="quickstart.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            TumorDecon
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="about.html">About TumorDecon</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quickstart</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Usage:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Data Inputs</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#reading-in-rna-sequence-data">Reading in RNA Sequence Data</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#read_rna_file"><code class="docutils literal notranslate"><span class="pre">read_rna_file()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#reading-in-signature-matrices-and-gene-sets">Reading in Signature Matrices and Gene Sets</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#read_sig_file"><code class="docutils literal notranslate"><span class="pre">read_sig_file()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#read_geneset"><code class="docutils literal notranslate"><span class="pre">read_geneset()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#creating-custom-lists-of-up-down-regulated-genes-from-signature-matrix">Creating Custom Lists of Up/Down-Regulated Genes from Signature Matrix</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#find_up_down_genes_from_sig"><code class="docutils literal notranslate"><span class="pre">find_up_down_genes_from_sig()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="sigmatrix.html">Signature Matrix Creation</a></li>
<li class="toctree-l1"><a class="reference internal" href="digital-cytometry.html">Running Digital Cytometry</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualization.html">Post-processing &amp; Visualization</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">TumorDecon</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Data Inputs</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/data-input.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="data-inputs">
<h1>Data Inputs<a class="headerlink" href="#data-inputs" title="Link to this heading"></a></h1>
<section id="reading-in-rna-sequence-data">
<h2>Reading in RNA Sequence Data<a class="headerlink" href="#reading-in-rna-sequence-data" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="read_rna_file">
<span class="sig-name descname"><span class="pre">read_rna_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">rna_file_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">identifier</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'hugo'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fetch_missing_hugo</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#read_rna_file" title="Link to this definition"></a></dt>
<dd><dl class="simple">
<dt>Read in a cbioportal pancancer or TCGA Xena Hub txt file containing mixture gene expression data, and return a Pandas DataFrame. Processing includes:</dt><dd><ul class="simple">
<li><p>removing genes with no identifier</p></li>
<li><p>removing duplicate genes (keep the first occurance)</p></li>
<li><p>removing genes with NaN values</p></li>
</ul>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>file_path</strong> – String. Relative or full path to the RNA Seq file. This file is tab seperated and includes two columns, ‘Hugo_Symbol’ and ‘Entrez_Gene_Id’, for each gene preceding the expression values for each patient</p></li>
<li><p><strong>identifier</strong> – String. Determines which gene identifier to use to index the rna data. Must be set to either:
- ‘hugo’ for Hugo Symbols
- ‘entrez’ for Entrez Gene ID</p></li>
<li><p><strong>fetch_missing_hugo</strong> – Boolean. Whether to fetch missing Hugo Symbols (by Entrez Gene ID) from ncbi website</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Pandas DataFrame, indexed by either Hugo_Symbol or Entrez_Gene_Id, containing RNA expression data for each patient. Rows are genes, columns are patients.</p>
</dd>
</dl>
</dd></dl>

<p>In addition to reading in an already-downloaded RNA dataset, you can fetch datasets directly from either <a class="reference external" href="https://www.cbioportal.org/">cBioPortal</a> or <a class="reference external" href="https://xena.ucsc.edu/">UCSC Xena</a> directly:</p>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">download_by_name(source,</span> <span class="pre">type,</span> <span class="pre">download_to=td.get_td_Home()+&quot;data/downloaded/&quot;,</span> <span class="pre">fetch_missing_hugo=False):</span></span></dt>
<dd><p>Downloads TCGA RNA Sequence data from cBioPortal and UCSC Xena Hub.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>source</strong> – String. Must be either “xena” or “cbio”. Where to download data from.</p></li>
<li><p><strong>type</strong> – String. Must match the specific identifier used for a cancer type by the given source</p></li>
<li><p><strong>download_to</strong> – String. Full or relative path to directory to save downloaded data to. Default is the data/downloads folder within the package’s library folder (OS/install dependent)</p></li>
<li><p><strong>fetch_missing_hugo</strong> – Boolean. Whether to fetch missing Hugo Symbols (by Entrez Gene ID) from ncbi website</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Pandas DataFrame, indexed by either Hugo_Symbol or Entrez_Gene_Id, containing RNA expression data for each patient. Rows are genes, columns are patients.</p>
</dd>
</dl>
</dd></dl>

<p>The current list of valid strings for <code class="docutils literal notranslate"><span class="pre">type</span></code>, as of March 2022, is the following:</p>
<p><strong>cBioPortal:</strong> ‘Acute Myeloid Leukemia’, ‘Adrenocortical Carcinoma’, ‘Bladder Urothelial Carcinoma’, ‘Brain Lower Grade Glioma’, ‘Breast Invasive Carcinoma’, ‘Cervical Squamous Cell Carcinoma’, ‘Cholangiocarcinoma’, ‘Colorectal Adenocarcinoma’, ‘Diffuse Large B-Cell Lymphoma’, ‘Esophageal Adenocarcinoma’, ‘Glioblastoma Multiforme’, ‘Head and Neck Squamous Cell Carcinoma’, ‘Kidney Chromophobe’, ‘Kidney Renal Clear Cell Carcinoma’, ‘Kidney Renal Papillary Cell Carcinoma’, ‘Liver Hepatocellular Carcinoma’, ‘Lung Adenocarcinoma’, ‘Lung Squamous Cell Carcinoma’, ‘Mesothelioma’, ‘Ovarian Serous Cystadenocarcinoma’, ‘Pancreatic Adenocarcinoma’, ‘Pheochromocytoma and Paraganglioma’, ‘Prostate Adenocarcinoma’, ‘Sarcoma’, ‘Skin Cutaneous Melanoma’, ‘Stomach Adenocarcinoma’, ‘Testicular Germ Cell Tumors’, ‘Thymoma’, ‘Thyroid Carcinoma’, ‘Uterine Carcinosarcoma’, ‘Uterine Corpus Endometrial Carcinoma’, ‘Uveal Melanoma’</p>
<p><strong>UCSC Xena:</strong> ‘Acute Myeloid Leukemia’, ‘Adrenocortical Cancer’, ‘Bile Duct Cancer’, ‘Bladder Cancer’, ‘Breast Cancer’, ‘Cervical Cancer’, ‘Colon and Rectal Cancer’, ‘Colon Cancer’, ‘Endometrioid Cancer’, ‘Esophageal Cancer’, ‘Glioblastoma’, ‘Head and Neck Cancer’, ‘Kidney Chromophobe’, ‘Kidney Clear Cell Carcinoma’, ‘Kidney Papillary Cell Carcinoma’, ‘Large B-cell Lymphoma’, ‘Liver Cancer’, ‘Lower Grade Glioma’, ‘Lower Grade Glioma and Glioblastoma’, ‘Lung Adenocarcinoma’, ‘Lung Cancer’, ‘Lung Squamous Cell Carcinoma’, ‘Melanoma’, ‘Mesothelioma’, ‘Ocular Melanomas’, ‘Ovarian Cancer’, ‘Pancreatic Cancer’, ‘Pheochromocytoma and Paraganglioma’, ‘Prostate Cancer’, ‘Rectal Cancer’, ‘Sarcoma’, ‘Stomach Cancer’, ‘Testicular Cancer’, ‘Thymoma’, ‘Thyroid Cancer’, ‘Uterine Carcinosarcoma’</p>
<p><strong>Examples</strong>:</p>
<div class="highlight-rst notranslate"><div class="highlight"><pre><span></span>&gt;&gt;&gt; import os
&gt;&gt;&gt; rna_cbio = td.download_by_name(&#39;cbio&#39;, &#39;Ovarian Serous Cystadenocarcinoma&#39;, download_to=os.getcwd(), fetch_missing_hugo=True)
Downloading data from cbioportal to /Users/&lt;username&gt;/ov_tcga_pan_can_atlas_2018.tar.gz...
100% [......................................................................] 103274157 / 103274157
Fetching missing Hugo Symbols for genes by Entrez ID...
Found  1 / 13 of the missing values
&gt;&gt;&gt; print(rna_cbio)
             TCGA-04-1348-01  TCGA-04-1357-01  TCGA-04-1362-01       ...         TCGA-OY-A56Q-01  TCGA-VG-A8LO-01  TCGA-WR-A838-01
Hugo_Symbol                                                          ...
UBE2Q2P2           25.716221        20.473921        29.909980       ...               15.716029        28.912917        59.400754
HMGB1P1           321.241331       128.865135       424.021806       ...              400.614342       309.174887       431.623999
LOC155060         222.150613       244.412320       377.449014       ...              387.776781       412.614342       540.597847
RNU12-2P            0.676632         0.090280         0.451162       ...                0.453561         7.464714         4.282079
EZHIP              -0.059851        -0.479632         4.514755       ...               -0.306245         0.693589         1.320945
<span class="cp">...                      ...              ...              ...       ...                     ...              ...              ...</span>
ZYG11A              6.824650         4.775693        66.800880       ...              141.004912         7.760542        79.382924
ZYG11B            483.942790       518.654716       791.336454       ...             1000.073852      1321.299392      1068.161532
ZYX              9692.870954      3284.754545      3747.937644       ...             6698.448007      4961.305257      3907.580348
ZZEF1             429.818654       684.474498       766.249800       ...              448.557587       709.015420      1665.424483
ZZZ3              475.233973       414.753874       802.717598       ...             2462.888278      1295.464299      1443.856175

[19064 rows x 300 columns]

&gt;&gt;&gt; rna_xena = td.download_by_name(&#39;xena&#39;, &#39;Ovarian Cancer&#39;, fetch_missing_hugo=False)
Downloading data from Xena Hub to /opt/py3.9env/lib/python3.9/site-packages/TumorDecon/data/downloaded//HiSeqV2.gz...
100% [........................................................................] 16904473 / 16904473
&gt;&gt;&gt; print(rna_xena)
              TCGA-61-1910-01  TCGA-61-1728-01  TCGA-09-1666-01       ...         TCGA-29-1702-01  TCGA-24-1417-01  TCGA-57-1585-01
Hugo_Symbol                                                           ...
ARHGEF10L          775.638814       638.278161       292.069698       ...              285.819639       388.964774      1091.183840
HIF3A                8.704138       163.905991        50.190351       ...             1966.434649       522.229474       510.680697
RNF17                0.000000         0.000000         0.000000       ...                0.000000         0.000000         0.000000
RNF10             5536.810611      3688.475448      2081.065875       ...             2204.056642      3539.408216      3186.036883
RNF11             1098.628128      1764.914771      2540.034063       ...             2478.786156      2102.389042      1811.543621
<span class="cp">...                       ...              ...              ...       ...                     ...              ...              ...</span>
PTRF              2672.103531      1843.865596      3230.078307       ...             2975.497567      4477.500339      8065.348376
BCL6B              105.418119        36.424211        41.130701       ...               28.153557       140.337452       245.065741
GSTK1             5985.512182      5203.702549      6308.893763       ...             1089.444025      4591.614580      2086.123131
SELP                 1.934267        15.729240         4.905667       ...                2.928014        57.688259       105.403367
SELS              1351.019271       691.226326       645.676332       ...             1017.549232       503.321528       551.526558

[20530 rows x 308 columns]
</pre></div>
</div>
</section>
<section id="reading-in-signature-matrices-and-gene-sets">
<h2>Reading in Signature Matrices and Gene Sets<a class="headerlink" href="#reading-in-signature-matrices-and-gene-sets" title="Link to this heading"></a></h2>
<p>The examples in this User Guide use mostly the LM22 signature matrix, or in the case of rank-based methods, an up-regulated gene set derived from LM22. These cell signatures are included in the package.
However, TumorDecon can also be run with any user-provided cell signatures.</p>
<dl class="py function">
<dt class="sig sig-object py" id="read_sig_file">
<span class="sig-name descname"><span class="pre">read_sig_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">delim</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'\\t'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geneID</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'Hugo_Symbol'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#read_sig_file" title="Link to this definition"></a></dt>
<dd><p>Read in a signature matrix (signature gene expression data for a number of different cell types) from a .csv or .txt file, and (if applicable) convert the gene identifiers from Entrez/Ensembl Gene ID to Hugo Symbols. Returns a Pandas DataFrame, indexed by ‘Hugo_Symbol’, where the rows are the genes and the columns are the cell types you wish to include. If no <code class="docutils literal notranslate"><span class="pre">file_path</span></code> provided, this function returns the LM22 signature matrix.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>file_path</strong> – String. Relative or full path to signature matrix file. The first row of the file should include the names of your cell types, and the first column should be either Hugo Symbols or Entrez Gene IDs. If no file_path given, LM22 is assumed.</p></li>
<li><p><strong>delim</strong> – String. Delimiter to use. Default is ‘\t’ (tab separated)</p></li>
<li><p><strong>geneID</strong> – String. Must be either “Hugo_Symbol”, “Ensembl_Gene_ID”, “Entrez_Gene_ID”. Describes how genes are labeled in the signature matrix file</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Pandas DataFrame containing signature expression values of each gene for each cell type. Columns are the cell types, rows are the genes (indexed by Hugo Symbol).</p>
</dd>
</dl>
</dd></dl>

<p><strong>Example</strong>: Assume we have a tab-separated file called “my_sig_matrix.txt”, where the genes are reference by Ensembl ID:</p>
<div class="highlight-rst notranslate"><div class="highlight"><pre><span></span>gene            CD8_subtype_1 CD8_subtype_2   CD4_subtype_1   CD4_subtype_2   B_cell_subtype_1        B_cell_subtype_2        NK_subtype_1    NK_subtype_2    mono_subtype_1  mono_subtype_2  Endothelial_subtype_1   Endothelial_subtype_2   Fibroblast_subtype_1    Fibroblast_subtype_2    Neutrophils_subtype_1   Neutrophils_subtype_2
ENSG00000000938       116.0001394     24.38016936     8.453628541     1.672547127     36.5573964      73.34069196     407.2145862     156.8132734     447.4100013     557.9327456     24.76138974     2.174048414     2.142543208     5.412057344     558.770835      739.358363
ENSG00000001167       22.50340273     0.75239339      15.05620142     12.07992823     10.26260586     8.947390096     10.62795065     12.4308104      9.912563566     13.39346071     7.472100959     22.07936334     7.783380993     2.073507549     46.75973        37.21335
ENSG00000002834       154.161176      36.456214       110.4616367     99.4175757      59.25913937     51.64127258     104.0716383     191.6722237     76.53034102     94.86365533     65.90175897     15.73233353     28.93766895     10.79387212     259.2520035     250.373201
<span class="cp">...             .......     .......</span>
</pre></div>
</div>
<p>We can read this into python with:</p>
<div class="highlight-rst notranslate"><div class="highlight"><pre><span></span>&gt;&gt;&gt; sig = td.read_sig_file(&quot;my_sig_matrix.txt&quot;, geneID=&#39;Ensembl_Gene_ID&#39;)
&gt;&gt;&gt; print(sig.head())
             CD8_subtype_1  CD8_subtype_2  CD4_subtype_1          ...            Fibroblast_subtype_2  Neutrophils_subtype_1  Neutrophils_subtype_2
Hugo_Symbol                                                       ...
FGR             116.000139      24.380169       8.453629          ...                        5.412057             558.770835             739.358363
NFYA             22.503403       0.752393      15.056201          ...                        2.073508              46.759730              37.213350
LASP1           154.161176      36.456214     110.461637          ...                       10.793872             259.252004             250.373201
TSR3             39.389183      48.998150      16.131921          ...                        0.371443               2.038300               3.715013
NADK             16.429699      12.887361       9.760582          ...                        1.503814             871.354356            1006.496017

[5 rows x 16 columns]
</pre></div>
</div>
<dl class="py function">
<dt class="sig sig-object py" id="read_geneset">
<span class="sig-name descname"><span class="pre">read_geneset</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">file_path</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#read_geneset" title="Link to this definition"></a></dt>
<dd><p>Read in a .csv file containing the up / down regulated genes for each cell type</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>file_path</strong> – String. Relative or full path to csv file. File should have columns named for each cell type, and rows must contain Hugo Symbols for genes to be considered as up (or down) regulated for that cell type. If not all cell types have the same number of up(down) regulated genes, excess rows in each column should be coded as “NaN”
- If no file_path given, the up-regulated gene set derived in Le et al. (2020) is assumed.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Dictionary. Keys are the column names of the given csv file (cell types), values are a list of up (or down) regulated genes for that cell type</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="creating-custom-lists-of-up-down-regulated-genes-from-signature-matrix">
<span id="creating-gene-set"></span><h2>Creating Custom Lists of Up/Down-Regulated Genes from Signature Matrix<a class="headerlink" href="#creating-custom-lists-of-up-down-regulated-genes-from-signature-matrix" title="Link to this heading"></a></h2>
<p>The following function provides a method for determining a list of up-regulated and down-regulated genes from a given signature matrix:</p>
<dl class="py function">
<dt class="sig sig-object py" id="find_up_down_genes_from_sig">
<span class="sig-name descname"><span class="pre">find_up_down_genes_from_sig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sig_df</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">down_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">up_cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_plots</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#find_up_down_genes_from_sig" title="Link to this definition"></a></dt>
<dd><p>Given a signature matrix, <code class="docutils literal notranslate"><span class="pre">`sig_df</span></code>, find a list of up-regulated and down-regulated genes for each cell type in the signature, using the following method:</p>
<blockquote>
<div><ul class="simple">
<li><p>Divide each gene expression value in Signature Matrix by the median value of the given gene across all cell types.</p></li>
<li><p>All genes with ratios below <code class="docutils literal notranslate"><span class="pre">down_cutoff</span></code> are considered “down-regulated genes” while all genes with ratios above <code class="docutils literal notranslate"><span class="pre">up_cutoff</span></code> are considered “up-regulated genes”.</p></li>
</ul>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sig_df</strong> – Pandas DataFrame. Signature matrix (rows are genes, columns are cell types) indexed by Hugo Symbol</p></li>
<li><p><strong>down_cutoff</strong> – Float. Value to use for the cutoff point as to what should be considered a “down-regulated gene”</p></li>
<li><p><strong>up_cutoff</strong> – Float. Value to use for the cutoff point as to what should be considered an “up-regulated gene”</p></li>
<li><p><strong>show_plots</strong> – Boolean. Whether to show a plot of the sorted signature matrix ratios for each cell type (Can be used to help choose the cutoff points).</p></li>
</ul>
</dd>
<dt class="field-even">Return up<span class="colon">:</span></dt>
<dd class="field-even"><p>Dictionary. Keys are cell types, values are a List of up-regulated genes for that cell type</p>
</dd>
<dt class="field-odd">Return down<span class="colon">:</span></dt>
<dd class="field-odd"><p>Dictionary. Keys are cell types, values are a List of down-regulated genes for that cell type</p>
</dd>
</dl>
</dd></dl>

<p><strong>Example:</strong></p>
<div class="highlight-rst notranslate"><div class="highlight"><pre><span></span>&gt;&gt;&gt; LM6 = td.read_sig_file(&#39;LM6.txt&#39;)
&gt;&gt;&gt; up_geneset_LM6, down_geneset_LM6 = td.find_up_down_genes_from_sig(LM6, down_cutoff=0.4, up_cutoff=4.0, show_plots=True)
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/subplots.png"><img alt="_images/subplots.png" src="_images/subplots.png" style="width: 800px; height: 400px;" />
</a>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="quickstart.html" class="btn btn-neutral float-left" title="Quickstart" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="sigmatrix.html" class="btn btn-neutral float-right" title="Signature Matrix Creation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright workshop participant.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>